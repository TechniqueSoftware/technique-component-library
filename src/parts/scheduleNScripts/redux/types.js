export const key = 'scheduleNScripts'

const FOLLOWUPS = {
  GET: `${key}/FOLLOWUPS/GET`,
  ERROR: `${key}/FOLLOWUPS/ERROR`,
  SUCCESS: `${key}/FOLLOWUPS/SUCCESS`,
  PERMISSION: `${key}/FOLLOWUPS/PERMISSION`,
  SET: `${key}/FOLLOWUPS/SET`,
  INNER_PERMISSIONS: `${key}/FOLLOWUPS/INNER_PERMISSIONS`,
  SET_INNER_PERMISSIONS: `${key}/FOLLOWUPS/SET_INNER_PERMISSIONS`,
  GET_FEATURE_PERMISSIONS: `${key}/FOLLOWUPS/GET_FEATURE_PERMISSIONS`
}

const TOUCHPOINT = {
  OPEN: `${key}/TOUCHPOINT/OPEN`,
  EDIT: `${key}/TOUCHPOINT/EDIT`,
  EDIT_TOUCHPOINT: `${key}/TOUCHPOINT/EDIT_TOUCHPOINT`,
  EDIT_SUCCESS: `${key}/TOUCHPOINT/EDIT_SUCCESS`,
  EDIT_ERROR: `${key}/TOUCHPOINT/EDIT_ERROR`,
  NEW: `${key}/TOUCHPOINT/NEW`,
  NEW_SUCCESS: `${key}/TOUCHPOINT/NEW_SUCCESS`,
  NEW_ERROR: `${key}/TOUCHPOINT/NEW_ERROR`,
  DELETE: `${key}/TOUCHPOINT/DELETE`,
  DELETE_SUCCESS: `${key}/TOUCHPOINT/DELETE_SUCCESS`,
  DELETE_ERROR: `${key}/TOUCHPOINT/DELETE_ERROR`,
  CLOSE: `${key}/TOUCHPOINT/CLOSE`
}

const SCRIPTS = {
  GET: 'scheduleNScripts/SCRIPTS/GET',
  SAVE: 'scheduleNScripts/SCRIPTS/SAVE',
  EDIT: 'scheduleNScripts/SCRIPTS/EDIT'
}

const SEGMENTS = {
  CLOSE: `${key}/SEGMENTS/CLOSE`,
  GET: `${key}/SEGMENTS/GET`,
  SET: `${key}/SEGMENTS/SET`,
  GET_ATTRIBUTES: `${key}/SEGMENTS/GET_ATTRIBUTES`,
  SET_ATTRIBUTES: `${key}/SEGMENTS/SET_ATTRIBUTES`,
  SET_DATA: `${key}/SEGMENTS/SET_DATA`,
  SAVE: `${key}/SEGMENTS/SAVE`,
  DELETE: `${key}/SEGMENTS/DELETE`,
  GET_LOCATIONS: `${key}/SEGMENTS/GET_LOCATIONS`,
  SET_LOCATIONS: `${key}/SEGMENTS/SET_LOCATIONS`,
  GET_LIST: `${key}/SEGMENTS/GET_LIST`,
  SET_LIST: `${key}/SEGMENTS/SET_LIST`,
  GET_COPY_PROCESS: `${key}/SEGMENTS/GET_COPY_PROCESS`,
  SET_COPY_PROCESS: `${key}/SEGMENTS/SET_COPY_PROCESS`,
  SET_ALERT: `${key}/SEGMENTS/SET_ALERT`,
  ERROR: `${key}/SEGMENTS/ERROR`
}

const SNACKBAR = {
  CLOSE: `${key}/SNACKBAR/CLOSE`,
  SET: `${key}/SNACKBAR/SET`
}

export default {
  FOLLOWUPS,
  TOUCHPOINT,
  SCRIPTS,
  SEGMENTS,
  SNACKBAR
}
