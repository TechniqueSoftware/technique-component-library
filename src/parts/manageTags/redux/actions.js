import TYPES from './types'

const DATA_ACTIONS = {
  SET: payload => ({ type: TYPES.DATA.SET, payload }),

  REQUEST_TAGS: payload => ({ type: TYPES.DATA.REQUEST_TAGS, payload }),
  SAVE_TAGS: payload => ({ type: TYPES.DATA.SAVE_TAGS, payload }),
  DELETE_TAGS: payload => ({ type: TYPES.DATA.DELETE_TAGS, payload })
}

export default {
  DATA: DATA_ACTIONS
}
