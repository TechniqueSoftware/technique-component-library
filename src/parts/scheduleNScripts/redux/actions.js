import TYPES from './types'

const FOLLOWUPS_ACTIONS = {
  GET: () => ({ type: TYPES.FOLLOWUPS.GET }),
  ERROR: payload => ({ type: TYPES.FOLLOWUPS.ERROR, payload }),
  SUCCESS: payload => ({ type: TYPES.FOLLOWUPS.SUCCESS, payload }),
  PERMISSION: payload => ({ type: TYPES.FOLLOWUPS.PERMISSION, payload }),
  SET: payload => ({ type: TYPES.FOLLOWUPS.SET, payload }),
  INNER_PERMISSIONS: payload => ({ type: TYPES.FOLLOWUPS.INNER_PERMISSIONS, payload }),
  SET_INNER_PERMISSIONS: payload => ({ type: TYPES.FOLLOWUPS.SET_INNER_PERMISSIONS, payload }),
  GET_FEATURE_PERMISSIONS: payload => ({ type: TYPES.FOLLOWUPS.GET_FEATURE_PERMISSIONS, payload })
}

const TOUCHPOINT_ACTIONS = {
  OPEN: payload => ({ type: TYPES.TOUCHPOINT.OPEN, payload }),
  EDIT: payload => ({ type: TYPES.TOUCHPOINT.EDIT, payload }),
  EDIT_TOUCHPOINT: payload => ({ type: TYPES.TOUCHPOINT.EDIT_TOUCHPOINT, payload }),
  EDIT_SUCCESS: payload => ({ type: TYPES.TOUCHPOINT.EDIT_SUCCESS, payload }),
  EDIT_ERROR: payload => ({ type: TYPES.TOUCHPOINT.EDIT_ERROR, payload }),
  NEW: payload => ({ type: TYPES.TOUCHPOINT.NEW, payload }),
  NEW_SUCCESS: payload => ({ type: TYPES.TOUCHPOINT.NEW_SUCCESS, payload }),
  NEW_ERROR: payload => ({ type: TYPES.TOUCHPOINT.NEW_ERROR, payload }),
  DELETE: payload => ({ type: TYPES.TOUCHPOINT.DELETE, payload }),
  DELETE_SUCCESS: payload => ({ type: TYPES.TOUCHPOINT.DELETE_SUCCESS, payload }),
  DELETE_ERROR: payload => ({ type: TYPES.TOUCHPOINT.DELETE_ERROR, payload }),
  CLOSE: () => ({ type: TYPES.TOUCHPOINT.CLOSE })
}

const SCRIPT_ACTIONS = {
  GET: () => ({ type: TYPES.SCRIPTS.GET }),
  SAVE: payload => ({ type: TYPES.SCRIPTS.SAVE, payload }),
  EDIT: payload => ({ type: TYPES.SCRIPTS.EDIT, payload })
}

const SEGMENTS_ACTIONS = {
  CLOSE: () => ({ type: TYPES.SEGMENTS.CLOSE }),
  GET: payload => ({ type: TYPES.SEGMENTS.GET, payload }),
  SET: payload => ({ type: TYPES.SEGMENTS.SET, payload }),
  GET_ATTRIBUTES: () => ({ type: TYPES.SEGMENTS.GET_ATTRIBUTES }),
  SET_ATTRIBUTES: payload => ({ type: TYPES.SEGMENTS.SET_ATTRIBUTES, payload }),
  SET_DATA: payload => (({ type: TYPES.SEGMENTS.SET_DATA, payload })),
  SAVE: payload => ({ type: TYPES.SEGMENTS.SAVE, payload }),
  DELETE: payload => ({ type: TYPES.SEGMENTS.DELETE, payload }),
  GET_LOCATIONS: () => ({ type: TYPES.SEGMENTS.GET_LOCATIONS }),
  SET_LOCATIONS: payload => ({ type: TYPES.SEGMENTS.SET_LOCATIONS, payload }),
  GET_LIST: payload => ({ type: TYPES.SEGMENTS.GET_LIST, payload }),
  SET_LIST: payload => ({ type: TYPES.SEGMENTS.SET_LIST, payload }),
  GET_COPY_PROCESS: payload => ({ type: TYPES.SEGMENTS.GET_COPY_PROCESS, payload }),
  SET_COPY_PROCESS: payload => ({ type: TYPES.SEGMENTS.SET_COPY_PROCESS, payload }),
  SET_ALERT: payload => ({ type: TYPES.SEGMENTS.SET_ALERT, payload }),
  ERROR: payload => ({ type: TYPES.SEGMENTS.ERROR, payload })
}

const SNACKBAR_ACTIONS = {
  CLOSE: () => ({ type: TYPES.SNACKBAR.CLOSE }),
  SET: payload => ({ type: TYPES.SNACKBAR.SET, payload })
}

export default {
  FOLLOWUPS: FOLLOWUPS_ACTIONS,
  TOUCHPOINT: TOUCHPOINT_ACTIONS,
  SCRIPTS: SCRIPT_ACTIONS,
  SEGMENTS: SEGMENTS_ACTIONS,
  SNACKBAR: SNACKBAR_ACTIONS
}
