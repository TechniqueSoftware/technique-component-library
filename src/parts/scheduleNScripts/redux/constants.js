export const TP_OPTIONS = {
  MENU: 'menu',
  NEW: 'new',
  EDIT: 'edit'
}

export const GROUP_NAMES = {
  PROSPECT: 'Prospect',
  ACTIVE_MEMBER: 'Active Member',
  INACTIVE_MEMBER: 'Inactive Member',
  PT_PROSPECT: 'PT Prospect',
  PT_MEMBER: 'PT Member',
  PT_OPT_OUT: 'PT Opt Out',
  PERSONAL_TRAINING: 'Personal Training'
}
