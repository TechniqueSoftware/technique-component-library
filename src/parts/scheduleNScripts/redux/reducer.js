import { produce } from 'immer'
import { TP_OPTIONS } from './constants'
import TYPES from './types'

export const initialState = {
  loading: false,
  error: '',
  snackbar: '',
  followups: [],
  canEditScripts: false,
  canEditSchedules: false,
  touchpointCrud: {
    action: '',
    data: {},
    loading: false
  },
  script: {},
  segments: [],
  segmentModalData: {
    isOpen: false,
    title: '',
    error: ''
  },
  segmentAttributes: {
    marketing: [],
    membership: []
  },
  innerPermissions: {
    automaticFollowUpAction: false,
    followUpSegmentsAction: false
  },
  segmentLocations: [],
  segmentList: [],
  copyInProcess: {},
  popupAlert: ''
}

const reducer = (state = initialState, { type, payload }) => produce(state, draft => {
  switch (type) {
    case TYPES.FOLLOWUPS.GET:
      draft.loading = true
      draft.followups = []
      draft.error = ''
      break

    case TYPES.FOLLOWUPS.ERROR:
      draft.loading = false
      draft.error = payload
      draft.followups = []
      break

    case TYPES.FOLLOWUPS.SUCCESS:
      draft.loading = false
      draft.error = ''
      draft.followups = payload.followups
      draft.canEditSchedules = payload.canEditSchedules
      draft.canEditScripts = payload.canEditScripts
      break

    case TYPES.FOLLOWUPS.PERMISSION:
      draft.canEditSchedules = payload.canEditSchedules
      draft.canEditScripts = payload.canEditScripts
      break

    case TYPES.FOLLOWUPS.SET:
      draft.followups = payload
      break

    case TYPES.FOLLOWUPS.SET_INNER_PERMISSIONS:
      draft.innerPermissions = payload
      break

    case TYPES.TOUCHPOINT.OPEN:
      draft.touchpointCrud = {
        action: payload.action,
        data: payload.data
      }
      break

    case TYPES.TOUCHPOINT.CLOSE:
      draft.touchpointCrud = { action: '', data: {}, loading: false }
      break

    case TYPES.TOUCHPOINT.NEW:
      draft.touchpointCrud.loading = true
      break

    case TYPES.TOUCHPOINT.NEW_SUCCESS:
      draft.followups = payload
      draft.touchpointCrud.loading = false
      draft.snackbar = 'NEW_SUCCESS'
      break

    case TYPES.TOUCHPOINT.NEW_ERROR:
      draft.touchpointCrud.loading = false
      draft.touchpointCrud.error = payload
      break

    case TYPES.TOUCHPOINT.DELETE:
      draft.touchpointCrud.loading = true
      break

    case TYPES.TOUCHPOINT.DELETE_SUCCESS:
      draft.followups = payload
      draft.touchpointCrud.loading = false
      draft.snackbar = 'DELETE_SUCCESS'
      break

    case TYPES.TOUCHPOINT.DELETE_ERROR:
      draft.touchpointCrud.loading = false
      draft.touchpointCrud.error = payload
      break

    case TYPES.TOUCHPOINT.EDIT:
      draft.touchpointCrud.action = TP_OPTIONS.EDIT
      break

    case TYPES.TOUCHPOINT.EDIT_TOUCHPOINT:
      draft.touchpointCrud.loading = true
      break

    case TYPES.TOUCHPOINT.EDIT_SUCCESS:
      draft.followups = payload.followups
      draft.touchpointCrud.loading = false
      draft.snackbar = payload.snackbar
      break

    case TYPES.TOUCHPOINT.EDIT_ERROR:
      draft.touchpointCrud.loading = false
      draft.touchpointCrud.error = payload
      break

    case TYPES.SNACKBAR.CLOSE:
      draft.snackbar = ''
      break

    case TYPES.SNACKBAR.SET:
      draft.snackbar = payload
      break

    case TYPES.SCRIPTS.EDIT:
      draft.touchpointCrud = {
        action: payload.action,
        data: payload.data
      }
      break

    case TYPES.SCRIPTS.SAVE:
      draft.script = {
        data: payload.data
      }
      break

    case TYPES.SEGMENTS.SET:
      draft.segments = payload
      break

    case TYPES.SEGMENTS.SET_DATA:
      draft.segmentModalData.title = payload.title
      draft.segmentModalData.isOpen = true
      draft.segmentModalData.followUpType = payload.followUpType
      draft.segmentModalData.followUpStatus = payload.followUpStatus
      break

    case TYPES.SEGMENTS.SET_ATTRIBUTES:
      draft.segmentAttributes = payload
      break

    case TYPES.SEGMENTS.CLOSE:
      draft.segmentModalData.isOpen = false
      draft.segmentModalData.error = ''
      break

    case TYPES.SEGMENTS.ERROR:
      draft.segmentModalData.error = payload
      break

    case TYPES.SEGMENTS.SET_LOCATIONS:
      draft.segmentLocations = payload
      break

    case TYPES.SEGMENTS.SET_LIST:
      draft.segmentList = payload
      break

    case TYPES.SEGMENTS.SET_ALERT:
      draft.popupAlert = payload
      break

    case TYPES.SEGMENTS.SET_COPY_PROCESS:
      draft.copyInProcess = payload
      break
  }
})

export default reducer
