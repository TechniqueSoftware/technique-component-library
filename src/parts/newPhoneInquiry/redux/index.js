import { createManager } from '@redux/injector'

import reducer from './reducer'
import { key } from './types'
import saga from './sagas'

export { default as REDUCER } from './reducer'
export { default as ACTIONS } from './actions'
export { default as SELECTORS } from './selectors'
export { key, default as TYPES } from './types'

export const Manager = createManager({
  name: 'NewPhoneInquiryManager',
  key,
  reducer,
  saga
})
