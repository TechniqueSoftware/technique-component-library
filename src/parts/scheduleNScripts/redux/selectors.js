import { createSelector } from 'reselect'
import { get } from 'lodash'

import { key } from './types'
import { initialState } from './reducer'

const reducer = state => get(state, key, initialState)

const getLoading = createSelector(
  reducer, state => get(state, 'loading', false)
)

const getError = createSelector(
  reducer, state => get(state, 'error', '')
)

const getSnackbar = createSelector(
  reducer, state => get(state, 'snackbar', '')
)

const getFollowups = createSelector(
  reducer, state => get(state, 'followups', [])
)

const getTouchpointCrud = createSelector(
  reducer, state => get(state, 'touchpointCrud', {})
)

const getSegments = createSelector(
  reducer, state => get(state, 'segments', [])
)

const getSegmentAttributes = createSelector(
  reducer, state => get(state, 'segmentAttributes', { marketing: [], membership: [] })
)

const getSegmentModal = createSelector(
  reducer, state => get(state, 'segmentModalData', { isOpen: false, title: '', error: '' })
)

const getCanEditScripts = createSelector(
  reducer, state => get(state, 'canEditScripts', false)
)

const getCanEditSchedules = createSelector(
  reducer, state => get(state, 'canEditSchedules', false)
)

const getInnerPermissions = createSelector(
  reducer, state => get(state, 'innerPermissions', {})
)

const getSegmentLocations = createSelector(
  reducer, state => get(state, 'segmentLocations', [])
)

const getSegmentList = createSelector(
  reducer, state => get(state, 'segmentList', [])
)

const getCopySegmentProcess = createSelector(
  reducer, state => get(state, 'copyInProcess', {})
)

const getPopupAlert = createSelector(
  reducer, state => get(state, 'popupAlert', '')
)

const getStatus = createSelector(
  [
    reducer,
    (_, { followUpType, followUpStatus }) => ({ followUpType, followUpStatus })
  ],
  (state, { followUpType, followUpStatus }) => {
    const followups = get(state, 'followups', [])

    const status = followups.reduce((acc, curr) => {
      if (curr.type !== followUpType) return acc
      return curr.statuses.find(s => s.id === `${followUpStatus}`) || acc
    }, null)
    return status
  }
)

export default {
  LOADING: getLoading,
  ERROR: getError,
  FOLLOWUPS: getFollowups,
  TOUCHPOINT_CRUD: getTouchpointCrud,
  CAN_EDIT_SCRIPTS: getCanEditScripts,
  CAN_EDIT_SCHEDULES: getCanEditSchedules,
  STATUS: getStatus,
  SEGMENTS: getSegments,
  SEGMENT_MODAL_DATA: getSegmentModal,
  SEGMENT_ATTRIBUTES: getSegmentAttributes,
  SNACKBAR: getSnackbar,
  INNER_PERMISSIONS: getInnerPermissions,
  SEGMENT_LOCATIONS: getSegmentLocations,
  SEGMENT_LIST: getSegmentList,
  SEGMENT_COPY_PROCESS: getCopySegmentProcess,
  POPUP_ALERT: getPopupAlert
}
