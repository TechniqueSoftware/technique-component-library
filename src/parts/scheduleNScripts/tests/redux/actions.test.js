import { ACTIONS } from '../../redux'

import { followupsMock, segmentsMock, locationIdMock, permissionMock } from './mock'

describe('ACTIONS', () => {
  describe('Followups', () => {
    it('gets', () => { expect(ACTIONS.FOLLOWUPS.GET()).toMatchSnapshot() })

    it('errors', () => {
      expect(ACTIONS.FOLLOWUPS.ERROR({
        error: 'some error here'
      })).toMatchSnapshot()
    })

    it('succeeds', () => {
      expect(ACTIONS.FOLLOWUPS.SUCCESS(followupsMock)).toMatchSnapshot()
    })

    it('sets permission', () => {
      expect(ACTIONS.FOLLOWUPS.PERMISSION({
        canEditSchedules: true,
        canEditScripts: false
      })).toMatchSnapshot()
    })

    it('sets', () => {
      expect(ACTIONS.FOLLOWUPS.SET(followupsMock)).toMatchSnapshot()
    })

    it('gets inner permissions', () => {
      expect(ACTIONS.FOLLOWUPS.INNER_PERMISSIONS(locationIdMock)).toMatchSnapshot()
    })

    it('sets inner permissions', () => {
      expect(ACTIONS.FOLLOWUPS.SET_INNER_PERMISSIONS(permissionMock)).toMatchSnapshot()
    })
  })

  describe('Touchpoint', () => {
    it('opens', () => {
      expect(ACTIONS.TOUCHPOINT.OPEN({
        payload: 'some id here'
      })).toMatchSnapshot()
    })

    it('edits', () => {
      expect(ACTIONS.TOUCHPOINT.EDIT({
        payload: 'some id here'
      })).toMatchSnapshot()
    })

    it('edits schedule', () => {
      expect(ACTIONS.TOUCHPOINT.EDIT_TOUCHPOINT({
        payload: { dayNum: 2, immediate: false }
      })).toMatchSnapshot()
    })

    it('edits errors', () => {
      expect(ACTIONS.TOUCHPOINT.EDIT_ERROR({
        payload: 'some error here'
      })).toMatchSnapshot()
    })

    it('edits succeeds', () => {
      expect(ACTIONS.TOUCHPOINT.EDIT_SUCCESS({
        payload: { followups: [], snackbar: 'EDIT_SUCCESS' }
      })).toMatchSnapshot()
    })

    it('creates', () => {
      expect(ACTIONS.TOUCHPOINT.NEW({
        payload: 'some id here'
      })).toMatchSnapshot()
    })

    it('creates errors', () => {
      expect(ACTIONS.TOUCHPOINT.NEW_ERROR({
        payload: 'some error here'
      })).toMatchSnapshot()
    })

    it('creates succeeds', () => {
      expect(ACTIONS.TOUCHPOINT.NEW_SUCCESS({
        payload: 'some id here'
      })).toMatchSnapshot()
    })

    it('deletes', () => {
      expect(ACTIONS.TOUCHPOINT.DELETE({
        payload: 'some id here'
      })).toMatchSnapshot()
    })

    it('deletes errors', () => {
      expect(ACTIONS.TOUCHPOINT.DELETE_ERROR({
        payload: 'some error here'
      })).toMatchSnapshot()
    })

    it('deletes succeeds', () => {
      expect(ACTIONS.TOUCHPOINT.DELETE_SUCCESS({
        payload: [{ a: 1 }, { a: 2 }]
      })).toMatchSnapshot()
    })

    it('closes', () => {
      expect(ACTIONS.TOUCHPOINT.CLOSE()).toMatchSnapshot()
    })
  })

  describe('Snackbar', () => {
    it('closes snackbar', () => {
      expect(ACTIONS.SNACKBAR.CLOSE()).toMatchSnapshot()
    })

    it('sets snackbar', () => {
      expect(ACTIONS.SNACKBAR.SET({ payload: 'Snackbar msg' })).toMatchSnapshot()
    })
  })

  describe('Scripts', () => {
    it('edits', () => {
      expect(
        ACTIONS.SCRIPTS.EDIT({
          payload: { action: 'some action', data: {} }
        })
      ).toMatchSnapshot()
    })

    it('saves', () => {
      expect(
        ACTIONS.SCRIPTS.SAVE({
          payload: { data: {} }
        })
      ).toMatchSnapshot()
    })
  })

  describe('Segments', () => {
    it('sets segments', () => {
      expect(ACTIONS.SEGMENTS.SET(segmentsMock)).toMatchSnapshot()
    })

    it('sets segments data', () => {
      expect(ACTIONS.SEGMENTS.SET_DATA({ title: 'title', followUpType: 1, followUpStatus: '11' })).toMatchSnapshot()
    })

    it('sets segments attributes', () => {
      expect(ACTIONS.SEGMENTS.SET_ATTRIBUTES({ marketing: ['a'], membership: ['b'] })).toMatchSnapshot()
    })

    it('closes segments modal', () => {
      expect(ACTIONS.SEGMENTS.CLOSE()).toMatchSnapshot()
    })

    it('sets error', () => {
      expect(ACTIONS.SEGMENTS.ERROR('Some error here')).toMatchSnapshot()
    })
  })
})
