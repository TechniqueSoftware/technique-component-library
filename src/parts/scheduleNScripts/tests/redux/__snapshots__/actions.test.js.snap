// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ACTIONS Followups errors 1`] = `
Object {
  "payload": Object {
    "error": "some error here",
  },
  "type": "scheduleNScripts/FOLLOWUPS/ERROR",
}
`;

exports[`ACTIONS Followups gets 1`] = `
Object {
  "type": "scheduleNScripts/FOLLOWUPS/GET",
}
`;

exports[`ACTIONS Followups gets inner permissions 1`] = `
Object {
  "payload": Object {
    "locationId": 4585,
  },
  "type": "scheduleNScripts/FOLLOWUPS/INNER_PERMISSIONS",
}
`;

exports[`ACTIONS Followups sets 1`] = `
Object {
  "payload": Object {
    "content": Array [
      Object {
        "clubId": 659,
        "followUpTypes": Array [
          Object {
            "followUpStatuses": Array [
              Object {
                "customClubLabel": null,
                "groupName": "Prospect",
                "hidden": "false",
                "id": "4",
                "name": "Active Guest",
              },
              Object {
                "customClubLabel": null,
                "groupName": "Prospect",
                "hidden": "false",
                "id": "5",
                "name": "Expired Guest",
              },
            ],
            "id": "1",
            "name": "Membership",
          },
          Object {
            "followUpStatuses": Array [
              Object {
                "customClubLabel": null,
                "groupName": "PT Prospect",
                "hidden": "false",
                "id": "1",
                "name": "Not Booked",
              },
            ],
            "id": "2",
            "name": "Personal Training",
          },
        ],
      },
    ],
  },
  "type": "scheduleNScripts/FOLLOWUPS/SET",
}
`;

exports[`ACTIONS Followups sets inner permissions 1`] = `
Object {
  "payload": Object {
    "innerPermissions": Object {
      "automaticFollowUpAction": true,
      "followUpSegmentsAction": false,
    },
  },
  "type": "scheduleNScripts/FOLLOWUPS/SET_INNER_PERMISSIONS",
}
`;

exports[`ACTIONS Followups sets permission 1`] = `
Object {
  "payload": Object {
    "canEditSchedules": true,
    "canEditScripts": false,
  },
  "type": "scheduleNScripts/FOLLOWUPS/PERMISSION",
}
`;

exports[`ACTIONS Followups succeeds 1`] = `
Object {
  "payload": Object {
    "content": Array [
      Object {
        "clubId": 659,
        "followUpTypes": Array [
          Object {
            "followUpStatuses": Array [
              Object {
                "customClubLabel": null,
                "groupName": "Prospect",
                "hidden": "false",
                "id": "4",
                "name": "Active Guest",
              },
              Object {
                "customClubLabel": null,
                "groupName": "Prospect",
                "hidden": "false",
                "id": "5",
                "name": "Expired Guest",
              },
            ],
            "id": "1",
            "name": "Membership",
          },
          Object {
            "followUpStatuses": Array [
              Object {
                "customClubLabel": null,
                "groupName": "PT Prospect",
                "hidden": "false",
                "id": "1",
                "name": "Not Booked",
              },
            ],
            "id": "2",
            "name": "Personal Training",
          },
        ],
      },
    ],
  },
  "type": "scheduleNScripts/FOLLOWUPS/SUCCESS",
}
`;

exports[`ACTIONS Scripts edits 1`] = `
Object {
  "payload": Object {
    "payload": Object {
      "action": "some action",
      "data": Object {},
    },
  },
  "type": "scheduleNScripts/SCRIPTS/EDIT",
}
`;

exports[`ACTIONS Scripts saves 1`] = `
Object {
  "payload": Object {
    "payload": Object {
      "data": Object {},
    },
  },
  "type": "scheduleNScripts/SCRIPTS/SAVE",
}
`;

exports[`ACTIONS Segments closes segments modal 1`] = `
Object {
  "type": "scheduleNScripts/SEGMENTS/CLOSE",
}
`;

exports[`ACTIONS Segments sets error 1`] = `
Object {
  "payload": "Some error here",
  "type": "scheduleNScripts/SEGMENTS/ERROR",
}
`;

exports[`ACTIONS Segments sets segments 1`] = `
Object {
  "payload": Array [
    Object {
      "createdAt": "2022-11-03T21:47:51Z",
      "createdBy": 20,
      "isDeleted": false,
      "name": "Test Segment",
      "segmentAttributes": Array [
        Object {
          "attributeId": 1936,
          "attributeType": "Marketing Source",
          "attributeValue": "Flyer",
        },
        Object {
          "attributeId": 1940,
          "attributeType": "Marketing Source",
          "attributeValue": "Other",
        },
      ],
      "segmentId": 1935,
      "updatedAt": "2022-11-03T22:01:10Z",
      "updatedBy": 20,
    },
    Object {
      "createdAt": "2022-11-07T16:58:04Z",
      "createdBy": 20,
      "isDeleted": false,
      "name": "Test Segment I",
      "segmentAttributes": Array [
        Object {
          "attributeId": 1944,
          "attributeType": "Marketing Source",
          "attributeValue": "Drive By",
        },
      ],
      "segmentId": 1943,
      "updatedAt": "2022-11-07T16:58:04Z",
      "updatedBy": 20,
    },
  ],
  "type": "scheduleNScripts/SEGMENTS/SET",
}
`;

exports[`ACTIONS Segments sets segments attributes 1`] = `
Object {
  "payload": Object {
    "marketing": Array [
      "a",
    ],
    "membership": Array [
      "b",
    ],
  },
  "type": "scheduleNScripts/SEGMENTS/SET_ATTRIBUTES",
}
`;

exports[`ACTIONS Segments sets segments data 1`] = `
Object {
  "payload": Object {
    "followUpStatus": "11",
    "followUpType": 1,
    "title": "title",
  },
  "type": "scheduleNScripts/SEGMENTS/SET_DATA",
}
`;

exports[`ACTIONS Snackbar closes snackbar 1`] = `
Object {
  "type": "scheduleNScripts/SNACKBAR/CLOSE",
}
`;

exports[`ACTIONS Snackbar sets snackbar 1`] = `
Object {
  "payload": Object {
    "payload": "Snackbar msg",
  },
  "type": "scheduleNScripts/SNACKBAR/SET",
}
`;

exports[`ACTIONS Touchpoint closes 1`] = `
Object {
  "type": "scheduleNScripts/TOUCHPOINT/CLOSE",
}
`;

exports[`ACTIONS Touchpoint creates 1`] = `
Object {
  "payload": Object {
    "payload": "some id here",
  },
  "type": "scheduleNScripts/TOUCHPOINT/NEW",
}
`;

exports[`ACTIONS Touchpoint creates errors 1`] = `
Object {
  "payload": Object {
    "payload": "some error here",
  },
  "type": "scheduleNScripts/TOUCHPOINT/NEW_ERROR",
}
`;

exports[`ACTIONS Touchpoint creates succeeds 1`] = `
Object {
  "payload": Object {
    "payload": "some id here",
  },
  "type": "scheduleNScripts/TOUCHPOINT/NEW_SUCCESS",
}
`;

exports[`ACTIONS Touchpoint deletes 1`] = `
Object {
  "payload": Object {
    "payload": "some id here",
  },
  "type": "scheduleNScripts/TOUCHPOINT/DELETE",
}
`;

exports[`ACTIONS Touchpoint deletes errors 1`] = `
Object {
  "payload": Object {
    "payload": "some error here",
  },
  "type": "scheduleNScripts/TOUCHPOINT/DELETE_ERROR",
}
`;

exports[`ACTIONS Touchpoint deletes succeeds 1`] = `
Object {
  "payload": Object {
    "payload": Array [
      Object {
        "a": 1,
      },
      Object {
        "a": 2,
      },
    ],
  },
  "type": "scheduleNScripts/TOUCHPOINT/DELETE_SUCCESS",
}
`;

exports[`ACTIONS Touchpoint edits 1`] = `
Object {
  "payload": Object {
    "payload": "some id here",
  },
  "type": "scheduleNScripts/TOUCHPOINT/EDIT",
}
`;

exports[`ACTIONS Touchpoint edits errors 1`] = `
Object {
  "payload": Object {
    "payload": "some error here",
  },
  "type": "scheduleNScripts/TOUCHPOINT/EDIT_ERROR",
}
`;

exports[`ACTIONS Touchpoint edits schedule 1`] = `
Object {
  "payload": Object {
    "payload": Object {
      "dayNum": 2,
      "immediate": false,
    },
  },
  "type": "scheduleNScripts/TOUCHPOINT/EDIT_TOUCHPOINT",
}
`;

exports[`ACTIONS Touchpoint edits succeeds 1`] = `
Object {
  "payload": Object {
    "payload": Object {
      "followups": Array [],
      "snackbar": "EDIT_SUCCESS",
    },
  },
  "type": "scheduleNScripts/TOUCHPOINT/EDIT_SUCCESS",
}
`;

exports[`ACTIONS Touchpoint opens 1`] = `
Object {
  "payload": Object {
    "payload": "some id here",
  },
  "type": "scheduleNScripts/TOUCHPOINT/OPEN",
}
`;
